presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -180,7 +180,25 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Void context)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Void context)
         {
-            return true;
+            /*
+             * Column reference can resolve to (a) some subquery's scope, (b) a projection (ORDER BY scope),
+             * (c) source scope or (d) outer query scope (effectively a constant).
+             * From AggregationAnalyzer's perspective, only case (c) needs verification.
+             */
+            return getReferencesToScope(node, analysis, sourceScope)
+                    .allMatch(this::verifySubqueryReferenceToSourceScope);
+        }
+
+        private boolean verifySubqueryReferenceToSourceScope(Expression referenceExpression)
```

Comment:
> Also, in a way, it's similar to "isColumnReferenceUseCorrect"

The latter got renamed as you suggested. It's now `isGroupingKey`.

I removed the `verifySubqueryReferenceToSourceScope` method, reusing `isGroupingKey` to find offending reference and inlining exc throwing to `visitSubqueryExpression`